impl/src/main/java/org/jboss/weld/resolution/ResolvableBuilder.java

```diff
@@ -50,7 +51,8 @@
 
 public class ResolvableBuilder {
 
-    private static final Class<?>[] FACADE_TYPES = new Class<?>[] { Event.class, Instance.class, WeldInstance.class, Provider.class, InterceptionFactory.class };
+    private static final Class<?>[] FACADE_TYPES = new Class<?>[] { Event.class, Instance.class, WeldEvent.class,
```

Comment:
Not really sure we need to change this?

Reply:
I believe we need this.

Reply:
I _think_ it's for when you resolve bean via BM (why would you do that with Event though?), but I am not really sure. If I remove it `@Inject` will keep working.